Sources/Quick/ExampleGroup.swift

```diff
@@ -8,6 +8,11 @@ import Foundation
 final public class ExampleGroup: NSObject {
     weak internal var parent: ExampleGroup?
     internal let hooks = ExampleHooks()
+    
+    internal var beforesStartedExecuting = false
```

Comment:
Modelling this state with 4 `bool`s creates the theoretical possibility of a bunch of invalid states. I know we were trying to move away from the large enum in the first iteration (sorry!) but I'd suggest using a smaller enum (5 cases, if I'm thinking about this correctly) here as the "source of truth". These particular booleans can then be computed properties on the enum, as necessary.